components/suggestion/generic_channel_provider.tsx

```diff
@@ -1,11 +1,23 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import {Channel} from '@mattermost/types/channels.js';
+import {ServerError} from '@mattermost/types/errors.js';
+import {ActionResult} from 'mattermost-redux/types/actions.js';
```

Comment:
Remove the `.js` extension on these imports. These are all `ts` files already, and we try to avoid file extensions in import since that causes extra work once we convert them to Typescript. You will see existing code using the file extension name, but that is something we are moving away from.